### DISCUSSION TITLE
Unexpected peak wave period

### DISCUSSION THREAD

**Discussion by newl74** [NOAA-EMC/WW3#1105] [Q&A]
Hello. I am the user of WW3 for making a forecast system. When I implement WW3 for the study area. I got a unexpected peak wave period. But, when I checked input features(wind, restart file, and so on), I couldn't find any other special issues. If someone has a similar experience like me, I wish to know what is the reason to happen this situation and how to solve this problem. I attached a figure for getting your opinions.

[Peak wave period problem.pdf](https://github.com/NOAA-EMC/WW3/files/13051598/Peak.wave.period.problem.pdf)

**Comment by ukmo-ccbunney** [comment]
Are you applying any boundary conditions along your southern edge?

**Reply by newl74** [reply]
Hello. All boundaries have boundary conditions. I try to run different scale of model, but I have a same problem at same point.

**Reply by aliabdolali** [reply]
Is it a curvilinear grid or unstructured mesh?
If it is at a certain point, for different scenarios, could you check the depth an gradient of depth in that area?

**Reply by aliabdolali** [reply]
Just as a check, could you mask that node and see if the problem persists? You can modify and define land elevation in the depth and it would be masked out.

**Reply by newl74** [reply]
I use regular rectangle grid now. Grid gradient is so smooth. Do you have any other suggestions to check my problem?

**Comment by mickaelaccensi** [comment]
I've already seen that on past version of WW3. Do you use the last version of develop branch ?

**Reply by newl74** [reply]
I use v.6.07. Which version of ww3 has same problem? I wish to know what was your problem and solution please.

**Reply by mickaelaccensi** [reply]
I don't remember when it was corrected but I remember having large peak period with very low Hs values. I advice you to try with the last develop branch version.

**Reply by newl74** [reply]
Thank you for good information. I will try it.

**Comment by aronroland** [comment]
@newl74, all, 

as I see things the ETOT goes down to 0, which then triggers the issue in the peak period. I guess there is some missing exception handling for this variable when ETOT lim -> 0.

**Reply by newl74** [reply]
I will try it. Thank you for your opinion.

**Reply by newl74** [reply]
Hello. I tried to find how to edit ETOT in WW3, but I didn't find that. If you know how to edit ETOT in model, please leave me an reply for me.

**Comment by aliabdolali** [comment]
As Mickael suggested, try the latest version of develop, if it is still there, share your setup with the community.

**Reply by newl74** [reply]
Thank you for your answer. I will try it.

**Reply by newl74** [reply]
Hello. Could you talk to me where I can find the latest version of develop? I have been searching where it is, but I didn't find.

**Reply by aliabdolali** [reply]
https://github.com/NOAA-emc/ww3/tree/develop

**Reply by newl74** [reply]
I got it. Thank you so much.
